extensions/warp-ipfs/src/store/document.rs

```diff
@@ -279,3 +284,93 @@ impl RootDocument {
         Ok(root_document)
     }
 }
+
+#[derive(Clone, Deserialize, Serialize)]
+pub struct FileAttachmentDocument {
+    pub name: String,
+    pub size: usize,
+    pub thumbnail: Cid,
+    pub file_type: FileType,
+    pub data: Cid,
+}
+
+impl FileAttachmentDocument {
+    pub async fn resolve_to_file(&self, ipfs: &Ipfs) -> Result<File, Error> {
+        let file = File::new(&self.name);
```

Comment:
is it a name or path? If former, why do we save in $PWD? 🤔

Reply:
This is the name of a file, that is recorded apart of [File](https://github.com/Satellite-im/Warp/blob/main/warp/src/constellation/file.rs) for constellation. Not related to the os filesystem itself

Reply:
(which is apart of the internal index)